x/application/keeper/transfer_applications.go

```diff
@@ -0,0 +1,241 @@
+
+import (
+	"context"
+	"fmt"
+
+	cosmostypes "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/pokt-network/poktroll/telemetry"
+	"github.com/pokt-network/poktroll/x/application/types"
+	"github.com/pokt-network/poktroll/x/shared"
+)
+
+// EndBlockerTransferApplication completes pending application transfers.
+// This always happens on the last block of a session during which the transfer started.
+// It is accomplished by:
+//  1. Copying the current state of the source app onto a new destination app
+//  2. Unstaking (removing) the source app
+//  3. Staking (storing) the destination app
+func (k Keeper) EndBlockerTransferApplication(ctx context.Context) error {
+	isSuccessful := false
+	defer telemetry.EventSuccessCounter(
+		"transfer_application_end",
+		telemetry.DefaultCounterFn,
+		func() bool { return isSuccessful },
+	)
+
+	sdkCtx := cosmostypes.UnwrapSDKContext(ctx)
+	currentHeight := sdkCtx.BlockHeight()
+	sharedParams := k.sharedKeeper.GetParams(ctx)
+	logger := k.Logger().With("method", "EndBlockerTransferApplication")
+
+	// Only process application transfers at the end of the session in
+	// order to avoid inconsistent/unpredictable mid-session behavior.
+	if !shared.IsSessionEndHeight(&sharedParams, currentHeight) {
+		return nil
+	}
+
+	// Iterate over all applications and transfer the ones that have finished the transfer period.
+	// TODO_MAINNET: Use an index to iterate over the applications that have initiated
+	// the transfer action instead of iterating over all of them.
+	for _, srcApp := range k.GetAllApplications(ctx) {
+		// Ignore applications that have not initiated the transfer action.
+		if !srcApp.HasPendingTransfer() {
+			continue
+		}
+
+		// Ignore applications that have initiated a transfer but still active.
+		// This spans the period from the end of the session in which the transfer
+		// began to the end of settlement for that session.
+		transferEndHeight := types.GetApplicationTransferHeight(&sharedParams, &srcApp)
+		if sdkCtx.BlockHeight() < transferEndHeight {
+			continue
+		}
+
+		// Transfer the stake of the source application to the destination application and
+		// merge their gateway delegations and service configs.
+		if transferErr := k.transferApplication(ctx, srcApp); transferErr != nil {
+			logger.Warn(transferErr.Error())
+
+			// Application transfer failed, removing the pending transfer from the source application.
+			dstBech32 := srcApp.GetPendingTransfer().GetDestinationAddress()
+			srcApp.PendingTransfer = nil
+			k.SetApplication(ctx, srcApp)
+
+			if err := sdkCtx.EventManager().EmitTypedEvent(&types.EventTransferError{
+				SourceAddress:      srcApp.GetAddress(),
+				DestinationAddress: dstBech32,
+				SourceApplication:  &srcApp,
+				Error:              transferErr.Error(),
+			}); err != nil {
+				return err
+			}
+		}
+	}
+
+	isSuccessful = true
+	return nil
+}
+
+// transferApplication transfers the fields of srcApp, except for address and pending_transfer,
```

Comment:
Can we remove 81-87?

Reply:
Not sure how that happened - thanks for calling it out! :raised_hands: